### DISCUSSION TITLE
Vertical & Horizontal = mixed up

### DISCUSSION THREAD

**Discussion by ophoria** [hako-mikan/sd-webui-regional-prompter#197] [Ideas]
In the translation into English the buttons vertical and horizontal are mixed up....

![image](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/75803107/d2fc2dc4-e34f-4601-9cc3-26dae39fe353)

**Comment by Symbiomatrix** [comment]
Yeah, it goes way back to when hako created the extension with 1D mode only, see [here](https://github.com/hako-mikan/sd-webui-regional-prompter#usage):
>This setting divides the image vertically into three parts and applies the prompts "green hair twintail" ,"red blouse" ,"blue skirt", from top to bottom in order.

I always preferred the rows / columns notation since it seems less confusing somehow (comma for columns, semicolon for rows, except for backwards compatibility with 1D with commas only), but I left the vertical / horizontal designations alone. Potential hassle to reexplain everything.

**Comment by hako-mikan** [comment]
This is an issue between languages. 
In Japanese, when we say "split horizontally", it means to split along a horizontal line. 
However, in English, it's expressed as splitting with a horizontal line. 
It seems there was a misunderstanding here.

**Reply by ophoria** [reply]
Thanks for teaching me :) ...I figured there was a good explanation, and there is.
With that said, it still makes little to no sense for most Europeans (and probably Americans), and since it's in English it might be an idea to change the English translation of it only, and just leave the original untouched?

**Comment by dayunbao** [comment]
It was confusing to me at first until I thought of it this way:
![horizontal_stacking](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/11196050/837c8dbf-d60b-4245-9ee4-e88f9f770484)

![vertical_stacking](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/11196050/c2a40523-4c82-4315-80f1-c5ba18280798)

The game engine I develop in has UI components that work this way.  So a "horizontal box container" will stacking things horizontally (from left to right).

I think tweaking the documentation to present it this way would alleviate a lot of confusion.  It also wouldn't require any changes to the UI or underlying code.  I'm willing to make the changes myself and open a PR.  What do you think @hako-mikanさん?

**Comment by hako-mikan** [comment]
I looked into it as well, and it seems this kind of issue existed previously. 
The Japanese person who addressed it at that time apparently couldn't find an appropriate expression. 
Therefore, it looks like I'll need to make a switch at some point. 
When doing so, I'm considering using the phrases 'split by Horizontal' and 'split by Vertical'. Would this wording be acceptable?

**Reply by dayunbao** [reply]
Do you want to avoid refactoring and swapping "vertical" with "horizontal" in the UI and code?  If so, you'd need to say something like, "New regions are added to the right horizontally when using comma-separated values," for Split mode Horizontal.  And "New regions are added to the bottom vertically when using comma-separated values," for Split mode Vertical.  That is a lot of words, though, and still might be confusing.

The other option would be to simply swap "vertical" and "horizontal."  Then it makes sense without any further need for explanation (to Western minds, anyway).

**Reply by hako-mikan** [reply]
I'm thinking of simply swapping them. 
I believe lengthy explanations aren't ideal. 
Besides, I recently released an extension with a similar function, and it's implemented in the correct direction, so, I'm thinking of making them consistent in terms of functionality."
https://github.com/hako-mikan/sd-webui-cd-tuner

**Reply by dayunbao** [reply]
> I'm thinking of simply swapping them.
I believe lengthy explanations aren't ideal.

I agree completely.

**Reply by dayunbao** [reply]
By the way, are newlines significant in the prompts?  I'm asking because the examples show ADDCOL being on the line before the prompt for a new column, and personally it makes more sense to me to have ADDCOL and the prompt for that column being on the same line.  (I haven't had a chance to look at the code where you parse the prompts yet, so I'm being lazy and asking you :bow:).  This is rather subjective, though.

**Comment by hako-mikan** [comment]
Line breaks are ignored, so you can write as you please without any issues.

**Reply by dayunbao** [reply]
Awesome.  Thanks!
